doc/GettingStarted.md

```diff
@@ -0,0 +1,225 @@
+# Getting Started
+
+On this page you will learn about [TSL](#TSL) and its Generator, what it is inteded for and how to utilize it.
+
+
+# Introduction
+
+### **What is TSL and what is it used for?**
+
+TSL is a C++ Header-only Library which abstracts elemental operations on 
+hardware specific instructions sets, espacial SIMD / vector operations.
+
+### **Why the indirection through a generator? Why no plain header only library?**
+
```

Comment:
It is a very hard, laborious, and maybe even impossible task to design a hardware abstraction library, which suits all currently available and upcoming hardware. So we didn't even try but decided to make it considerably simple to change nearly every detail of the library provided interface and implementation.
To enable such flexibility, we wrote a generator that relies on Jinja2 -- a powerful and mature template engine.
Furthermore,...